LibVLCSharp/Platforms/UWP/VideoView.cs

```diff
@@ -0,0 +1,233 @@
+        /// be displayed in your application.
+        /// </summary>
+        /// <returns>The list of arguments to be given to the <see cref="LibVLC"/> constructor.</returns>
+        public string[] GetSwapChainOptions()
+        {
+            if (!_loaded)
+            {
+                throw new InvalidOperationException("You must wait for the VideoView to be loaded before calling GetSwapChainOptions()");
+            }
+
+            return new string[]
+            {
+                $"--winrt-d3dcontext=0x{_d3d11Device.ImmediateContext.NativePointer.ToString("x")}",
+                $"--winrt-swapchain=0x{_swapChain.NativePointer.ToString("x")}"
+            };
+        }
+
+        void OnLoad()
+        {
+            SharpDX.DXGI.Factory2 dxgiFactory;
+            DeviceCreationFlags deviceCreationFlags = DeviceCreationFlags.BgraSupport | DeviceCreationFlags.VideoSupport;
+
+#if DEBUG
+            deviceCreationFlags |= DeviceCreationFlags.Debug;
+            try
+            {
+                dxgiFactory = new SharpDX.DXGI.Factory2(true);
+            }
+            catch (SharpDXException e)
+            {
+                dxgiFactory = new SharpDX.DXGI.Factory2(false);
+            }
+#else
+            dxgiFactory = new SharpDX.DXGI.Factory2(false);
+#endif
+            _d3d11Device = null;
+            foreach (var adapter in dxgiFactory.Adapters)
+            {
+                try
+                {
+                    _d3d11Device = new SharpDX.Direct3D11.Device(adapter, deviceCreationFlags);
+                    break;
+                }
+                catch (SharpDXException e)
+                {
+                }
+            }
+
+            if (_d3d11Device is null)
+            {
+                throw new Exception("Could not D3D11CreateDevice");
+            }
+
+            _device = _d3d11Device.QueryInterface<SharpDX.DXGI.Device1>();
+
+            //Create the swapchain
+            var swapChainDescription = new SharpDX.DXGI.SwapChainDescription1
+            {
+                Width = (int)(this._panel.ActualWidth * this._panel.CompositionScaleX),
+                Height = (int)(this._panel.ActualHeight * this._panel.CompositionScaleY),
+                Format = SharpDX.DXGI.Format.B8G8R8A8_UNorm,
+                Stereo = false,
+                SampleDescription =
+                {
+                    Count = 1,
+                    Quality = 0
+                },
+                Usage = Usage.RenderTargetOutput,
+                BufferCount = 2,
+                SwapEffect = SwapEffect.FlipSequential,
+                Flags = SwapChainFlags.None,
+                AlphaMode = AlphaMode.Unspecified
+            };
+
+            _swapChain = new SharpDX.DXGI.SwapChain1(dxgiFactory, _d3d11Device, ref swapChainDescription);
+
+            _device.MaximumFrameLatency = 1;
+
+            //TODO: perform the next 2 calls on the UI thread
```

Comment:
This is a legacy of the original code. What we can do:

- Leave as-is, and we need to investigate later
- We are already on the main thread because we handled the Loaded event, and this is fine, and we can remove the comment
- All the other calls must be done on another thread, and maybe we should switch threads here

Reply:
> We are already on the main thread because we handled the Loaded event, and this is fine, and we can remove the comment

Our users might not run this code from the Loaded event. Unsure if that's an issue here...

Reply:
Nope this is our code called on load that creates the swapchain, and before the user actually creates the libvlc instance

Reply:
From the beginning of the method to this line, the code takes 23ms on my computer to run.

Would switching threads bring any benefit there? I think that would only make the code harder to read. I'll remove the //TODO.